### ISSUE TITLE
'yield-star-spacing' rule conflict

### ISSUE THREAD

**Issue by xianghongai** [Microsoft/vscode-eslint#630]
.eslintrc.json

```json
"extends": [ "standard", "prettier", "prettier/standard" ],
"rules": { 
    "generator-star-spacing": [ "warn", "before" ], 
    "yield-star-spacing": [ "warn", "before" ]
}
```

yield-star-spacing.js
```javascript
/* eslint yield-star-spacing: ["error", "after"] */
function* generator() { yield* other(); }

/* eslint yield-star-spacing: ["error", "before"] */
function *generator2() { yield *other(); }

/* eslint yield-star-spacing: ["error", "both"] */
function *generator3() { yield * other(); }

/* eslint yield-star-spacing: ["error", "neither"] */
function*generator4() { yield*other(); }

function other () {}
```

![2019-02-22-171323](https://user-images.githubusercontent.com/2075006/53232079-4aa07e00-36c5-11e9-98fd-b77e651a5316.png)

**Comment by dbaeumer** [comment]
The same problems are reported when running eslint in the terminal:

<img width="512" alt="capture" src="https://user-images.githubusercontent.com/1931590/53415737-41b7f100-39d2-11e9-9993-2dfc4e12ac4a.png">

Please report the issue to ESLint itself if you think there is something wrong with the reported diagnostics.
